src/server/node_func.c

```diff
@@ -2377,7 +2379,13 @@ decode_Mom_list(struct attribute *patr, char *name, char *rescn, char *val)
 
 	for (i = 0; (p = str_arr[i]) != NULL; i++) {
 		clear_attr(&new, &node_attr_def[(int)ND_ATR_Mom]);
-		if (get_fullhostname(p, buf, (sizeof(buf) - 1)) != 0) {
+		is_node_name_ip = inet_pton(AF_INET, p, &(check_ip.sin_addr)) ;
+		if(!is_node_name_ip) {
+			if (get_fullhostname(p, buf, (sizeof(buf) - 1)) != 0) {
+				strncpy(buf, p, (sizeof(buf) - 1));
+				buf[sizeof(buf) - 1] = '\0';
```

Comment:
Since we are doing the above 2 steps in both if and else part, is it possible to take them out of the if...else condition?

Reply:
Hey @npadole20 I think that wouldn't be correct, as we can see the purpose of that part of code is different here. That mean setting of buffer "when we don't get the full hostname of a node which we added using node name" vs "when we added node using ip address".

Reply:
I think I agree with Neha, you can convert this to:
```
if (if_node_name_ip || get_fullhostname(p, buf, (sizeof(buf) - 1)) == 0) {
    strncpy(buf, p, (sizeof(buf) - 1));
    buf[sizeof(buf) - 1] = '\0';
}
```

right?

Reply:
Yes agree with both of you. Thanks, I will make the change